sematic/cli/clean.py

```diff
@@ -49,16 +47,15 @@ def clean(orphaned_jobs: bool, orphaned_resources: bool, force: bool):
     Clean up objects that are no longer needed.
     """
     running_as_cron_job = os.environ.get("RUNNING_AS_CRON_JOB", None) is not None
-    switch_env("user")
     if running_as_cron_job:
-        dictConfig(make_log_config(log_to_disk=False))
         echo = logger.info  # type: ignore
         api_key = get_cleaner_api_key()
         os.environ[UserSettingsVar.SEMATIC_API_KEY.value] = api_key
         echo("Starting cron cleaner from cron job")
     else:
         echo = click.echo  # type: ignore
         echo("Starting cleaner")
+    switch_env("user")
```

Comment:
Please set `SEMATIC_SERVER_ADDRESS` instead of `SEMATIC_API_ADDRESS` in `cleaner-cron.yaml`, and revert this move. [This code](https://github.com/sematic-ai/sematic/blob/f4ea69007780656420e9d75cb16f579b5dc50b71/sematic/config/config.py#L188-L189) should ensure the server URL is picked up correctly.

The `switch_env` should actually come before `running_as_cron_job = os.environ.get("RUNNING_AS_CRON_JOB", None) is not None`.

Reply:
I think `SEMATIC_API_ADDRESS` is actually more appropriate. `SEMATIC_SERVER_ADDRESS` is used for "what address do I start the server on", whereas `SEMATIC_API_ADDRESS` is "what address do I use to reach the API." This use case is the latter.

Why does `switch_env` need to come before `running_as_cron_job = os.environ.get("RUNNING_AS_CRON_JOB", None) is not None`? I have verified that this does indeed work for both locally executed CLI and running in the cloud.

Reply:
`switch_env` mangles env vars, so it should be the second thing that is done in any process execution, after the gevent monkeypatching. If you source env vars, then switch envs, you might get unintended values for the env vars.

Reply:
It also *uses* env vars, which is why I moved it down here. I can try it higher though and see if it still works.